plugins/formal-verification/formver.uniqueness/src/org/jetbrains/kotlin/formver/UniqueCheckerContext.kt

```diff
@@ -7,11 +7,21 @@ package org.jetbrains.kotlin.formver
 
 import org.jetbrains.kotlin.fir.FirSession
 import org.jetbrains.kotlin.fir.declarations.FirDeclaration
+import org.jetbrains.kotlin.fir.declarations.FirVariable
+import org.jetbrains.kotlin.fir.symbols.impl.FirVariableSymbol
+import org.jetbrains.kotlin.name.ClassId
+
+// TODO: find a better name for it
+sealed class PathUnique
+data class Path(val symbol: FirVariableSymbol<FirVariable>) : PathUnique()
+data class Level(val level: Set<UniqueLevel>) : PathUnique()
 
 interface UniqueCheckerContext {
     val config: PluginConfiguration
     val errorCollector: ErrorCollector
     val session: FirSession
+    val uniqueId: ClassId
+    val uniqueStack: ArrayDeque<ArrayDeque<PathUnique>>
```

Comment:
This is a confusing type to work with.  Could you make a dedicated class for this, please?  Preferably both the stack and its elements, I suspect.

Why is it a Deque if you want to use it as a stack?